Etg.Yams.sln

```diff
@@ -55,6 +55,10 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Etg.Yams.Ipc", "src\Etg.Yam
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Etg.Yams.Host", "src\Etg.Yams.Host\Etg.Yams.Host.csproj", "{C9A993F7-E3E2-4FDC-AA1E-01348B023D16}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "NugetFeedDeploymentRepository", "src\NugetFeedDeploymentRepository\NugetFeedDeploymentRepository.csproj", "{1110E183-76B6-4C9F-939A-1B9290C3FC72}"
```

Comment:
Are we going to create a NuGet for this repository? I am also wondering if we should create an Yams.Extensions solution (or even potentially a GitHub repo).

Reply:
Created a nuget package and fluent YamsServiceHost builder.

Reply:
@jkonecki I am very confused about the pattern you are using for fluent builder, which I haven't seen before. Usually, you create a builder class (e.g. YamsHostBuilder) that takes all required parameters in the constructor. Then you add fluent interface for optional configuration (take a look at YamsConfigBuilder). For external modules, they just add extension method to YamsHostBuilder.

Reply:
I removed the fluent interface implementation and created an overload for static `Create` method that accepts individual interfaces. 
I feel a contradiction in your requirements: on one hand you say that the constructor should take all required parameters and the extension methods should be for optional ones, and on the other hand you suggest creating an extension method for registering NuGet repository. But deployment repository is not an optional parameter (I cannot construct Yams service without one).

Reply:
@jkonecki You're right, sorry about that. I didn't notice that all parameters were required when I suggested a builder as an option. What you have right now looks pretty good. 

There is a common variation of the builder pattern where no parameters are given in the constructor and the Build() method does the work of making sure that all required parameters have been given. This is especially useful in languages where named parameters are not supported (for readability) or when one need to construct the builder in multiple steps (pass it around). I believe ASP.Net opted for this variation. I'd happy with this variation if you prefer.

Sorry again for the confusion!